dask/highlevelgraph.py

```diff
@@ -211,6 +147,77 @@ def get_dependencies(self, key: Hashable, all_hlg_keys: Iterable) -> set:
         """
         return keys_in_tasks(all_hlg_keys, [self[key]])
 
+    def __dask_distributed_anno_pack__(self) -> Optional[Mapping[str, Any]]:
+        """Packs Layer annotations for transmission to scheduler
+
+        Callables annotations are fully expanded over Layer keys, while
+        other values are simply transmitted as is
+
+        Returns
+        -------
+        packed_annotations : dict
+            Packed annotations.
+        """
+        if self.annotations is None:
+            return None
+
+        packed = {}
+
+        for a, v in self.annotations.items():
+            if callable(v):
+                packed[a] = {stringify(k): v(k) for k in self}
+                packed[a]["__expanded_annotations__"] = True
+            else:
+                packed[a] = v
+
+        return packed
+
+    @staticmethod
+    def __dask_distributed_anno_unpack__(
+        annotations: MutableMapping[str, Any],
+        new_annotations: Optional[Mapping[str, Any]],
+        keys: Iterable[Hashable],
+    ) -> None:
+        """
+        Unpack a set of layer annotations across a set of keys, then merge those
+        expanded annotations for the layer into an existing annotations mapping.
+
+        This is not a simple shallow merge because some annotations like retries,
+        priority, workers, etc need to be able to retain keys from different layers.
+
+        Parameters
+        ----------
+        annotations: MutableMapping[str, Any], output
+            Already unpacked annotations, which are to be updated with the new
+            unpacked annotations
+        keys: Iterable
+            All keys in the layer.
+        """
+        if new_annotations is None:
+            return
+
+        expanded = {}
+        keys_stringified = False
+
+        # Expand the new annotations across the keyset
+        for a, v in new_annotations.items():
+            if type(v) is dict and "__expanded_annotations__" in v:
+                # Maybe do a destructive update for efficiency?
+                v = v.copy()
+                del v["__expanded_annotations__"]
+                expanded[a] = v
+            else:
+                if not keys_stringified:
+                    keys = [stringify(k) for k in keys]
+                    keys_stringified = True
+
+                expanded[a] = dict.fromkeys(keys, v)
+
+        # Merge the expanded annotations with the existing annotations mapping
+        for k, v in expanded.items():
+            v.update(annotations.get(k, {}))
+        annotations.update(expanded)
```

Comment:
```suggestion
            else:
                if not keys_stringified:
                    keys = [stringify(k) for k in keys]
                    keys_stringified = True
                v = dict.fromkeys(keys, v)

            v.update(annotations.get(a, {}))
            annotations[a] = v
```

Reply:
In case of collision, the original annotations silently win; is it deliberate?

Reply:
I think this may be a case of the original annotations (i.e. supplied as kwargs to Client.{submit, persist} overriding layer annotations. @ian-r-rose may know more here.

Reply:
> I think this may be a case of the original annotations (i.e. supplied as kwargs to Client.{submit, persist} overriding layer annotations

Yes, this was also what I was thinking. @ian-r-rose what do you think?

Reply:
> I think this may be a case of the original annotations (i.e. supplied as kwargs to Client.{submit, persist} overriding layer annotations.

Sorry for the slow reply here. Yes, that was my understanding as well. *However*, I don't have a strong intuition of which *should* win, and I can think of an argument for both (more specific annotation vs later-applied annotation).